ifsbench/env.py

```diff
@@ -0,0 +1,197 @@
+    """
+
+    def __init__(self, mode, key=None, value=None):
+        if key is not None:
+            self._key = str(key)
+        else:
+            self._key = None
+
+        if value is not None:
+            self._value = str(value)
+        else:
+            self._value = None
+
+        self._mode = mode
+
+        if self._key is None and self._mode != EnvOperation.CLEAR:
+            raise ValueError(f"The key must be specified for operation {mode.name}!")
+
+        if self._value is None:
+            if self._mode in (EnvOperation.APPEND, EnvOperation.PREPEND):
+                raise ValueError(f"The value must be specified for operation {mode.name}!")
+
+    def execute(self, env):
+        """
+        Apply the stored changes to a given environment.
+
+        Parameters
+        ----------
+        env: dict
+            An environment dictionary. Will be updated in place.
+        """
+
+        if self._mode == EnvOperation.SET:
+            debug(f"Set environment entry {self._key} = {self._value}.")
+            env[self._key] = self._value
+        elif self._mode == EnvOperation.APPEND:
+            if env.get(self._key, None) is not None:
+                env[self._key] += ":" + self._value
+            else:
+                env[self._key] = self._value
+
+            debug(f"Append {self._value} to environment variable {self._key}.")
+        elif self._mode == EnvOperation.PREPEND:
+            if env.get(self._key, None) is not None:
+                env[self._key] = self._value + ':' + env[self._key]
+            else:
+                env[self._key] = self._value
+
+            debug(f"Prepend {self._value} to environment variable {self._key}.")
+
+        elif self._mode == EnvOperation.DELETE:
+            if self._key in env:
+                debug(f"Delete environment variable {str(self._key)}.")
+                del env[self._key]
+
+        elif self._mode == EnvOperation.CLEAR:
+            debug('Clear whole environment.')
+            env.clear()
+
+class EnvPipeline(ABC, UserList):
+    """
+    Abstract base class for environment update pipelines.
+
+    Instances of this class can be used to update environment variables.
+    EnvPipeline supports all list operations to add/remove EnvHandler objects.
+    """
+
+    @abstractmethod
+    def execute(self):
+        """
+        Create an environment using the pipeline.
+
+        Returns
+        -------
+        dict:
+            The created environment.
+        """
+        return NotImplemented
+
+    @abstractmethod
+    def add(self, handler):
+        """
+        Add new EnvHandlers to the pipeline.
+
+        Parameters
+        ----------
+        handler: EnvHandler or list[EnvHandler]
+            Handler(s) that are added.
+        """
+
+class DefaultEnvPipeline(EnvPipeline):
+    """
+    Default environment pipeline.
+
+    Parameters
+    ----------
+    overrides: list[EnvHandler]
```

Comment:
This looks like an outdated docstring
```suggestion
    handlers: list[EnvHandler]
```